boot.js

```diff
@@ -187,11 +203,45 @@ function assertPlugin (plugin) {
 
 // load a plugin
 Boot.prototype.use = function (plugin, opts) {
-  this._addPlugin(plugin, opts, false)
-
+  this._lastUsed = this._addPlugin(plugin, opts, false)
   return this
 }
 
+Boot.prototype._loadRegistered = function (plugin) {
+  plugin = plugin || this._lastUsed
+  return new Promise((resolve) => {
+    if (plugin && !plugin.loaded) {
+      plugin.asyncQ.push((s) => {
+        // TODO check for override support
+        resolve()
+      })
+    } else {
+      resolve()
+    }
+
+    // if the root plugin is not loaded, let's resume that
+    // so one can use after() befor calling ready
+    if (!this.started && !this.booted) {
+      this._root.q.resume()
+    }
+  })
+}
+
+Object.defineProperty(Boot.prototype, 'then', {
+  get () {
+    // If the instance is ready, then there is
+    // nothing to await. This is true during
+    // await server.ready() as ready() resolves
+    // with the server, end we will end up here
+    // because of automatic promise chaining.
+    if (this.booted) {
+      return undefined
```

Comment:
we don't need to explicitly return `undefined` - also is there a way to reuse this for server.then?

Reply:
done, see `thenify` function